config/config.go

```diff
@@ -88,46 +104,127 @@ var defaultConfig = &config{
 	},
 }
 
+// NewDefaultConfig returns a deep copy of the default configuration
+//
+// By doing this, we ensure that newConfig has its own independent copy of the MeteoProviders slice. Therefore, any changes made to newConfig will not affect defaultConfig.
+func NewDefaultConfig() *config {
+	newConfig := *defaultConfig
```

Comment:
Correct me if I'm wrong but doesn't this create a deep copy? The next 2 lines seem unnecessary..

Reply:
Ignore this i overlooked that MeteoProviders is a slice